superset/assets/src/chart/Chart.jsx

```diff
@@ -81,6 +85,17 @@ class Chart extends React.PureComponent {
         this.props.chartKey,
       );
     }
+
+    this.visPromise = visPromiseLookup[this.props.vizType];
```

Comment:
I want to verify that this works properly when changing a vis in explorer, may need to move it out of `componentDidMount`

Reply:
confirmed this breaks, will fix it.